server/weaverbird/backends/pandas_executor/steps/utils/cast.py

```diff
@@ -6,10 +6,13 @@ def cast_to_float(s: Series) -> Series:
 
 
 def cast_to_int(s: Series) -> Series:
-    s = cast_to_float(s)
+    s_float = cast_to_float(s)
     # Since .astype(int) would fail on NaN values, treat them separately:
-    s_integer = s[s.notna()].astype(int)
-    s_nan = s[s.isna()]
+    s_integer = s_float[s_float.notna()].astype(int)
+    s_nan = s_float[s_float.isna()]
+    # Pandas operate on NanoSecond Timestamp but we want to output MilliSecond Timestamps
+    if s.dtype == 'datetime64[ns]':
```

Comment:
Maybe we should us a startswith here, because if dates would arrive with timezone info, it would be reflected here I think
(https://pandas.pydata.org/docs/reference/api/pandas.DatetimeTZDtype.html#pandas.DatetimeTZDtype)